src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> t = tail;
+        Node<T> newNode = new Node<>(t, value, null);
+        tail = newNode;
+        if (t == null) {
+            head = newNode;
+        } else {
+            t.next = newNode;
+        }
+        size += 1;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> temp = head;
+            Node<T> newNode = new Node<>(null, value, temp);
+            temp.prev = newNode;
+            head = newNode;
+        } else {
+            Node<T> temp = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(temp.prev, value, temp);
+            temp.prev.next = newNode;
+            temp.prev = newNode;
+        }
+        size += 1;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T item : list) {
+            add(item);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> temp = getNodeByIndex(index);
+        T oldItem = temp.item;
+        temp.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> temp = getNodeByIndex(index);
+        Node<T> prev = temp.prev;
+        Node<T> next = temp.next;
+        if (prev == null) {
+            head = next;
+        } else {
+            prev.next = next;
+        }
+        if (next == null) {
+            tail = prev;
+        } else {
+            next.prev = prev;
+        }
+        size -= 1;
```

Comment:
let's move this code at separate method `unlink(Node<> nodeToRemove)` and let's use it at method remove(T t)